File: app/src/main/java/pl/adriankremski/coolector/repository/SessionRepositoryImpl.java
package pl.adriankremski.coolector.repository;

import android.content.Context;

import javax.inject.Inject;

import pl.adriankremski.coolector.TheApp;

public class SessionRepositoryImpl implements SessionRepository{

    @Inject
    Session session;

    public SessionRepositoryImpl(Context context) {
        TheApp.get(context).getAppComponent().inject(this);
    }

    @Override
    public boolean isLoggedIn() {
        return session.isLoggedIn();
    }

    @Override
    public void setSessionToken(String token) {
        session.setSessionToken(token);
    }

    @Override
    public void setSessionKey(String sessionKey) {
        session.setSessionKey(sessionKey);
    }

    @Override
    public String getSessionToken() {
        return session.getSessionToken();
    }
}

File: app/src/main/java/com/noordwind/apps/collectively/data/repository/SessionRepositoryImpl.kt
package com.noordwind.apps.collectively.data.repository

import com.noordwind.apps.collectively.data.datasource.Session
import com.noordwind.apps.collectively.domain.repository.SessionRepository

class SessionRepositoryImpl(val session: Session) : SessionRepository {

    override val isLoggedIn: Boolean
        get() = session.isLoggedIn

    override fun setSessionKey(sessionKey: String) {
        session!!.setSessionKey(sessionKey)
    }

    override var sessionToken: String?
        get() = session.sessionToken
        set(token) {
            session.setSessionToken(token)
        }
}
